spinoso-string/src/enc/utf8/mod.rs

```diff
@@ -0,0 +1,1060 @@
+                // multiple codepoints.
+                for ch in ch.to_lowercase() {
+                    replacement.push_char(ch);
+                }
+                bytes = &bytes[size..];
+            } else {
+                let (substring, remainder) = bytes.split_at(size);
+                replacement.extend_from_slice(substring);
+                bytes = remainder;
+            }
+        }
+        self.inner = replacement;
+    }
+
+    #[inline]
+    pub fn make_lowercase(&mut self) {
+        // This allocation assumes that in the common case, lower-casing
+        // `char`s do not change the length of the `String`.
+        let mut replacement = Vec::with_capacity(self.len());
+        let mut bytes = self.inner.as_slice();
+
+        while !bytes.is_empty() {
+            let (ch, size) = bstr::decode_utf8(bytes);
+            if let Some(ch) = ch {
+                // Converting a UTF-8 character to lowercase may yield
+                // multiple codepoints.
+                for ch in ch.to_lowercase() {
+                    replacement.push_char(ch);
+                }
+                bytes = &bytes[size..];
+            } else {
+                let (substring, remainder) = bytes.split_at(size);
+                replacement.extend_from_slice(substring);
+                bytes = remainder;
+            }
+        }
+        self.inner = replacement;
+    }
+
+    #[inline]
+    pub fn make_uppercase(&mut self) {
+        // This allocation assumes that in the common case, upper-casing
+        // `char`s do not change the length of the `String`.
+        let mut replacement = Vec::with_capacity(self.len());
+        let mut bytes = self.inner.as_slice();
+
+        while !bytes.is_empty() {
+            let (ch, size) = bstr::decode_utf8(bytes);
+            if let Some(ch) = ch {
+                // Converting a UTF-8 character to lowercase may yield
+                // multiple codepoints.
+                for ch in ch.to_uppercase() {
+                    replacement.push_char(ch);
+                }
+                bytes = &bytes[size..];
+            } else {
+                let (substring, remainder) = bytes.split_at(size);
+                replacement.extend_from_slice(substring);
+                bytes = remainder;
+            }
+        }
+        self.inner = replacement;
+    }
+}
+
+impl Utf8String {
+    #[inline]
+    #[must_use]
+    pub fn chr(&self) -> &[u8] {
+        match bstr::decode_utf8(self.inner.as_slice()) {
+            (Some(_), size) => &self.inner[..size],
+            (None, 0) => &[],
+            (None, _) => &self.inner[..1],
+        }
+    }
+
+    #[inline]
+    pub fn ord(&self) -> Result<u32, OrdError> {
+        let (ch, size) = bstr::decode_utf8(self.inner.as_slice());
+        match ch {
+            // All `char`s are valid `u32`s
+            // https://github.com/rust-lang/rust/blob/1.48.0/library/core/src/char/convert.rs#L12-L20
```

Comment:
this comment is a holdover from when this `From` impl didn't exist. Let's delete it now since the `u32::from` call makes this super clear.

```suggestion
```

Reply:
I assume you still wanted the first line of the comment, however let me know if you want that gone too!